Xamarin.Forms.Platform.iOS/Renderers/PageRenderer.cs

```diff
@@ -326,45 +356,98 @@ void IAccessibilityElementsController.ResetAccessibilityElements()
 			Container?.ClearAccessibilityElements();
 		}
 
+		bool ShouldUseSafeArea()
+		{
+			bool usingSafeArea = Page.On<PlatformConfiguration.iOS>().UsingSafeArea();
+			bool isSafeAreaSet = Element.IsSet(PageSpecific.UseSafeAreaProperty);
+
+			if (IsPartOfShell && !isSafeAreaSet)
+				usingSafeArea = true;
+
+			return usingSafeArea;
+		}
+
 		void UpdateUseSafeArea()
 		{
-			if (!Forms.IsiOS11OrNewer)
+			if (Element == null)
+				return;
+
+			if (_userOverriddenSafeArea)
+				return;
+
+			if (!IsPartOfShell && !Forms.IsiOS11OrNewer)
```

Comment:
I think there is a mistake. Because the method goes deeper only if `IsPartOfShell` and `Forms.IsiOS11OrNewer` must be `false`, but logic below expects one of them to be `true`.

Reply:
I don't follow. The if statement here is meant to maintain backwards compatibility because currently if you're not part of shell and you're running iOS10 then none of the safearea code will run so this is just a way to exit. 

This probably means I could remove this if statement though
https://github.com/xamarin/Xamarin.Forms/pull/6457/files#diff-d95827e5633c20beb139e10ccd1a40d9R401

but I don't think it's totally necessary to